src-input/duk_hobject_props.c

```diff
@@ -5663,7 +5663,7 @@ duk_bool_t duk_hobject_define_property_helper(duk_context *ctx,
 			a = (duk_harray *) obj;
 			DUK_DD(DUK_DDPRINT("Object.defineProperty() attribute update for duk_harray .length -> %02lx", (unsigned long) new_flags));
 			DUK_ASSERT_HARRAY_VALID(a);
-			if ((new_flags & DUK_PROPDESC_FLAGS_EC) != (curr.flags & DUK_PROPDESC_FLAGS_EC)) {
+			if ((new_flags & DUK_PROPDESC_FLAGS_EC) != ((duk_small_uint_t)curr.flags & DUK_PROPDESC_FLAGS_EC)) {
```

Comment:
- I'm not sure I understand the issue here: `curr.flags` should already be a duk_small_uint_t (duk_propdesc defines in duk_hobject.h)? What's the warning fixed?
- If this is needed, there should be a space after the cast (code style).